toke.c

```diff
@@ -6099,6 +6099,16 @@ yyl_caret(pTHX_ char *s)
 {
     char *d = s;
     const bool bof = cBOOL(FEATURE_BITWISE_IS_ENABLED);
+    if (s[1] == '^') {
+        s += 2;
+        if (!PL_lex_allbrackets && PL_lex_fakeeof >=
+                (*s == '=' ? LEX_FAKEEOF_ASSIGN : LEX_FAKEEOF_LOGIC)) {
+            s -= 2;
+            TOKEN(0);
+        }
+        pl_yylval.ival = OP_XOR;
```

Comment:
If  `ival` is now significant for `OROR` ops, you need to change:
```
    DEBUG_TOKEN (NONE,  OROR),
```
at the top of the file to match, so the `ival` is output under `-DT`, compare:
```
tony@venus:.../git/perl6$ ./perl -DT -e '$x = ($y or $z)' 2>&1 | grep OROP
### <== OROP(ival=180)
tony@venus:.../git/perl6$ ./perl -DT -e '$x = ($y ^^ $z)' 2>&1 | grep OROR
### <== OROR
09752913e8bd1235c96a5944a507c887aa46fe8c (HEAD -> 21996-log-xor) Create a new logical xor operator, spelled `^^`
```

Reply:
Ahyes. Now fixed + forcepushed.

```
leo@hevva:~/src/bleadperl/perl [git]
$ ./perl -DT -e '$x = ($y ^^ $z)' 2>&1 | grep OROR
### <== OROR(ival=181)

leo@hevva:~/src/bleadperl/perl [git]
$ ./perl -DT -e '$x = ($y || $z)' 2>&1 | grep OROR
### <== OROR(ival=180)
```